### PR TITLE
Add appstream metadata

### PR DESCRIPTION
**Pull request by androidnisse**
https://www.freedesktop.org/wiki/Distributions/AppStream/

Needed for this application to show up in various Linux software centers such as gnome-software and discover.

### CODE REVIEW

File: Runtime/org.codelite.codelite.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.codelite.codelite</id>
+  
+  <name>CodeLite</name>
+  <summary>CodeLite is an open source IDE for the C//C++ languages</summary>
```
Comment (Jarod42) [inline Runtime/org.codelite.codelite.metainfo.xml:6]: From Readme.md
> CodeLite is a free, open source, cross platform IDE specialized in C, C++, PHP and JavaScript (mainly for backend developers using Node.js) programming languages, which runs best on all major platforms (Windows, macOS and Linux).

Wonder if double `/` is intended

File: Runtime/org.codelite.codelite.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">codelite.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://screenshots.debian.net/shrine/screenshot/simage/large-e7feb090ac27fb2959d53b522e07c84e.jpg</image>
+    </screenshot>
+    <screenshot>
+      <image>https://screenshots.debian.net/shrine/screenshot/simage/large-5ba5845aec9fa7a591d7ab0f5e6a9605.jpg</image>
```
Comment (Jarod42) [inline Runtime/org.codelite.codelite.metainfo.xml:23]: Wonder if diff-view and database query are the best/more representative screenshots of the application ;-)

Comment (eranif) [reply Runtime/org.codelite.codelite.metainfo.xml:23]: I could upload to some new screenshots to https://codelite.org and we could use them instead

File: Runtime/org.codelite.codelite.metainfo.xml (hunk 1 above)
Comment (androidnisse) [reply Runtime/org.codelite.codelite.metainfo.xml:6]: Thanks for the feedback, I copied the summary from the Solus package. I will add the one from the Readme.md as it is better to be consistent.

File: Runtime/org.codelite.codelite.metainfo.xml (hunk 2 above)
Comment (androidnisse) [reply Runtime/org.codelite.codelite.metainfo.xml:23]: Yes, to be honest I just package this in Solus so I think it is better as you say to add new better pictures, just ping me here and I will use them instead.

**Comment by eranif** [comment]
I have uploaded some recent screenshots for CodeLite (macOS / Windows / Linux):


https://codelite.org/downloads/codelite/screenshots/rust-code-completion.png
https://codelite.org/downloads/codelite/screenshots/chat-ai-preview.png
https://codelite.org/downloads/codelite/screenshots/codelite-macos.png
https://codelite.org/downloads/codelite/screenshots/codelite-windows.png
https://codelite.org/downloads/codelite/screenshots/git-integration.png
https://codelite.org/downloads/codelite/screenshots/codelite-linux.png
https://codelite.org/downloads/codelite/screenshots/codelite-linux-2.png

**Comment by Eric-01** [comment]
While you are at it, it may also be a good idea to unify the CodeLite description itself. There are at least 13 (!) different variations of it, and they differ quite a bit. Some of them may even be no longer in use, and yet they are still in the repository.

Here's a list of the ones I managed to find:

---

> CodeLite is an open source, free, cross platform IDE, specialized in C, C++, Rust, Python, PHP and JavaScript (mainly for backend developers using Node.js) programming languages which runs best on all major Platforms ( OSX, Windows and Linux )

source: https://codelite.org/

---

> CodeLite is a free, open source, cross platform IDE specialized in C, C++, PHP and JavaScript (mainly for backend developers using Node.js) programming languages, which runs best on all major platforms (Windows, macOS and Linux).

sources: https://docs.codelite.org/ and https://github.com/eranif/codelite/blob/master/README.md

---

> CodeLite is an open-source, cross plattform IDE specialized in C, C++, PHP and JavaScript (mainly for backend developers using Node.js) programming languages which runs best on all major Plattforms (OSC, Windows and Linux)

source: https://community.chocolatey.org/packages/codelite

---

> CodeLite, a cross platform C/C++/Rust/PHP and Node.js IDE written in C++

source: https://matrix.to/#/#eranif_codelite:gitter.im

---

> A multi purpose IDE specialized in C/C++/Rust/Python/PHP and Node.js. Written in C++

source: https://github.com/eranif/codelite (the repository description in the sidebar)

---

> CodeLite, an Open Source cross platform C, C++, Rust, Python, PHP and Node.js IDE

source: https://github.com/eranif/codelite/blob/master/LiteEditor/code_parser.rc

---

> CodeLite, a free, open source, C/C++/Rust/Python/PHP and JavaScript IDE

source: https://github.com/eranif/codelite/blob/master/Plugin/clAboutDialogBase.wxcp

---

> C/C++/Rust/Python/PHP/Node.js IDE (Integrated Development Environment)

and

> C/C++ IDE (Integrated Development Environment)

source: https://github.com/eranif/codelite/blob/master/CMakeLists.txt

---

> CodeLite is an open source, free, cross platform IDE specialized in C, C++, PHP and JavaScript (mainly for backend developers using Node.js) programming languages

source: https://github.com/eranif/codelite/blob/master/DESC (last updated 9 YEARS ago)

---

> codelite - a lightweight and powerful C/C++ IDE

source: https://github.com/eranif/codelite/blob/master/Runtime/man1/codelite.1 (last updated 10 YEARS ago)

---

> CodeLite is a powerful open-source, cross platform code editor for C/C++

source: https://github.com/eranif/codelite/blob/master/codelite.spec (last updated 14 YEARS ago)

---

> CodeLite is a wxWidgets C++ code parser library for parsing C/C++ (and other languages supported by Ctags in the future) intended for using CodeCompletion in IDEs like Code::Blocks.

source: https://github.com/eranif/codelite/blob/master/CodeLite/DoxySMain.page (last updated 17 YEARS ago)

**Review by Jarod42 — APPROVED** [review]

**Pull request merged**